README.md

```diff
@@ -8,18 +9,45 @@ const archive = getAHyperdriveSomehow();
 
 const rawPath = '/blog/about'
 
-resolveDatPath(archive, rawPath, (err, resolution) => {
-  if(err) console.log('Show your application 404 page')
-  else {
-    const {type, path, stat} = resolution
-
-    if(type === 'directory') {
-      console.log('Render the file list from the folder signified by `path`')
-    } else if(type === 'file') {
-      console.log('Render the file at `path`')
-    } else {
-      console.error('Something went horribly wrong')
-    }
+try {
+  const {type, path, stat} = await resolveDatPath(archive, rawPath)
+  if(type === 'directory') {
+    console.log('Render the file list from the folder signified by `path`')
+  } else if(type === 'file') {
+    console.log('Render the file at `path`')
+  } else {
+    console.error('Something went horribly wrong')
   }
-})
+} catch (err) {
+  console.log('Show your application 404 page')
+}
 ```
+
+## How it works
+
+_History:_ This Algorithm is based on the work used by hashbase and [beaker][].
+
+[beaker]: https://github.com/beakerbrowser/hashbase/blob/2dc67348a6607fb75b9857779e5faf77348d449a/lib/apis/archive-files.js#L91-L172
+
+How the algorithm for looking up paths works:
+
+1. It will look for the `web_root` property in the `/dat.json` file to use as prefix, if non-existent it will use `/`.
```

Comment:
Since dat is deprecated, how do you feel about removing the `dat.json` and `web_root` stuff to reduce the noise? I don't think Beaker implements [webroot](https://docs.beakerbrowser.com/developers/index.json-manifest) anymore either.

I'm also okay with keeping it in there and figuring that out later.